pkg/transparentproxy/config/dataplane/config_dataplane.go

```diff
@@ -0,0 +1,164 @@
+package dataplane
+
+import (
+	"github.com/asaskevich/govalidator"
+	"golang.org/x/exp/constraints"
+
+	core_config "github.com/kumahq/kuma/pkg/config"
+	xds_types "github.com/kumahq/kuma/pkg/core/xds/types"
+	tproxy_config "github.com/kumahq/kuma/pkg/transparentproxy/config"
+	"github.com/kumahq/kuma/pkg/util/pointer"
+)
+
+type PortLike interface {
+	constraints.Integer | constraints.Float | tproxy_config.Port
```

Comment:
Why `Float`? 🤔

Reply:
When converting to `map[string]any` for `structpb.Struct`, numbers are stored as `float64` and stay that way when converting back.